File: src/codes/tight-coupling/py/uncode.py
class Mailer:
    def send(self, email: str, content: str):
        print(f"送信しました: {email} → {content}")

class UserService:
    def __init__(self):
        self.mailer = Mailer()  # ← 密結合：直接生成

    def notify_user(self, email: str):
        content = "ようこそ！"
        self.mailer.send(email, content)  # ← Mailer に強く依存

# 利用例
service = UserService()
service.notify_user("user@example.com")

File: src/codes/tight-coupling/php/uncode.php
<?php

class Mailer {
  public function send(string $email, string $content): void {
    echo "送信しました: {$email} → {$content}\n";
  }
}

class UserService {
  private Mailer $mailer;

  public function __construct() {
    $this->mailer = new Mailer(); // ← 密結合：直接生成
  }

  public function notifyUser(string $email): void {
    $content = "ようこそ！";
    $this->mailer->send($email, $content); // ← Mailer の存在を知りすぎている
  }
}

// 利用例
$service = new UserService();
$service->notifyUser("user@example.com");
